bokehjs/test/unit/core/bokeh_events.ts

```diff
@@ -61,3 +64,88 @@ describe("BokehEvent", () => {
     expect(event.values).to.be.equal({target: widget, action: "do_that"})
   })
 })
+
+describe("AxisClick event", () => {
+  for (const {side, pt} of [{side: "left", pt: xy(5, 200)}, {side: "right", pt: xy(395, 200)}, {side: "above", pt: xy(200, 5)}, {side: "below", pt: xy(200, 395)}]) {
+    it(`should support linear axes on side: ${side}`, async () => {
+      const plot = new Plot({
+        width: 400,
+        height: 400,
+        title: null,
+        toolbar_location: null,
+        x_range: new Range1d({start: 0, end: 10}),
+        y_range: new Range1d({start: 0, end: 10}),
+      })
+      const axis = new LinearAxis()
+      plot.add_layout(axis, side as any)
+
+      const events: AxisClick[] = []
+      axis.on_event(AxisClick, (event: any) => events.push(event))
+
+      const {view} = await display(plot)
+
+      const actions = new PlotActions(view, {units: "screen"})
+      await paint()
+      await actions.tap(pt)
```

Comment:
It doesn't make sense to call `await paint()` at this point, because `PlotActions` doesn't perform any actions during construction. It would make more sense to call it after `tap()`. However, given how the plot and the test are set up, there should be no rendering calls, thus nothing to await for and thus this should actually hang. Perhaps there are some unnecessary rendering being performed.

Reply:
This also applies to commonly used `await view.ready`, because no view related signals are being emitted on this code path.